### PR TITLE
PX4: startup more drivers on demand

### PR DESCRIPTION
**Pull request by tridge**
startup rangefinder and PWM drivers on demand when use selects the option rather than always starting

### CODE REVIEW

File: libraries/AP_RangeFinder/AP_RangeFinder_PX4.cpp [hunk 1]
```diff
@@ -75,11 +76,35 @@ AP_RangeFinder_PX4::~AP_RangeFinder_PX4()
     }
 }
 
+extern "C" {
+    int ll40ls_main(int, char **);
+    int trone_main(int, char **);
+    int mb12xx_main(int, char **);
+    int pwm_input_main(int, char **);
```
Comment (OXINARF) [inline libraries/AP_RangeFinder/AP_RangeFinder_PX4.cpp]: Is the *pwm_input_main* really needed in this file?

Comment (tridge) [inline libraries/AP_RangeFinder/AP_RangeFinder_PX4.cpp]: thanks, fixed

**Comment by tridge** [comment]
travis is failing with out of space error

**Comment by OXINARF** [comment]
I wrote in Gitter, I'm aware of it and waiting an answer from support. I'll have a fix tomorrow. It should compile all boards with make, so as long as that is working then it should be good to go.

**Comment by tridge** [comment]
applied

**Pull request closed without merging**